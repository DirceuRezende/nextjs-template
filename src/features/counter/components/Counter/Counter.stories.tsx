import { store } from '@/store';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { Provider } from 'react-redux';

import Counter from '.';

export default {
  title: 'Counter',
  component: Counter,
  decorators: [
    (Story) => (
      <Provider store={store}>
        <Story />
      </Provider>
    )
  ]
} as ComponentMeta<typeof Counter>;

const Template: ComponentStory<typeof Counter> = () => (
  <div>
    <Counter />
  </div>
);

export const FirstStory = Template.bind({});

FirstStory.args = {};
