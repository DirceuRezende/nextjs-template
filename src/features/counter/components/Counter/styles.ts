import styled from 'styled-components';

export const Wrapper = styled.section`
  width: 100%;
`;

export const Counter = styled.span`
  color: #f1f1f1;
  font-size: 15rem;
`;

export const CounterButton = styled.button`
  background-color: #010101;
  border: none;
  border-radius: 100%;
  color: #f1f1f1;
  cursor: pointer;
  font-size: 5rem;
  height: 7rem;
  width: 7rem;
`;

export const CounterButtonGroup = styled.div`
  display: flex;
  justify-content: center;
  gap: 5rem;
`;
