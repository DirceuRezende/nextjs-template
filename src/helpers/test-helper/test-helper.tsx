import { store } from '@/store';
import theme from '@/styles/theme';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';

const customRender = (child: React.ReactElement, options?: RenderOptions) =>
  render(
    <Provider store={store}>
      <ThemeProvider theme={theme}>{child}</ThemeProvider>
    </Provider>,
    options
  );

export * from '@testing-library/react';
export { customRender as render };
