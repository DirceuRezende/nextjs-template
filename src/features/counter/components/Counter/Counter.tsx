import { useAppDispatch, useAppSelector } from '@/store';
import Link from 'next/link';
import { decrement, increment } from '../../store';
import * as S from './styles';

const Counter = () => {
  const count = useAppSelector((state) => state.counter.value);
  const dispatch = useAppDispatch();
  return (
    <S.Wrapper>
      <S.Counter>{count}</S.Counter>
      <S.CounterButtonGroup>
        <S.CounterButton onClick={() => dispatch(decrement())}>
          -
        </S.CounterButton>
        <S.CounterButton onClick={() => dispatch(increment())}>
          +
        </S.CounterButton>
      </S.CounterButtonGroup>
      <Link href="/">Go Home</Link>
    </S.Wrapper>
  );
};

export default Counter;
