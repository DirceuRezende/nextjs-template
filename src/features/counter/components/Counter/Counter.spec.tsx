import { render, screen } from '@/helpers/test-helper';

import Counter from '.';

describe('<Counter />', () => {
  it('should render the heading', () => {
    const { container } = render(<Counter />);

    expect(screen.getByRole('link', { name: /Go Home/i })).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
