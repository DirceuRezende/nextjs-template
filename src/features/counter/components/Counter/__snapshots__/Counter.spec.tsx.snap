// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> should render the heading 1`] = `
.c0 {
  width: 100%;
}

.c1 {
  color: #f1f1f1;
  font-size: 15rem;
}

.c3 {
  background-color: #010101;
  border: none;
  border-radius: 100%;
  color: #f1f1f1;
  cursor: pointer;
  font-size: 5rem;
  height: 7rem;
  width: 7rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 5rem;
}

<section
  class="c0"
>
  <span
    class="c1"
  >
    0
  </span>
  <div
    class="c2"
  >
    <button
      class="c3"
    >
      -
    </button>
    <button
      class="c3"
    >
      +
    </button>
  </div>
  <a
    href="/"
  >
    Go Home
  </a>
</section>
`;
